src/Kestrel.Transport.Sockets/Internal/SocketReceiver.cs

```diff
@@ -17,9 +18,12 @@ public SocketAwaitableEventArgs WaitForDataAsync()
         {
             _awaitableEventArgs.SetBuffer(Array.Empty<byte>(), 0, 0);
 
-            if (!_socket.ReceiveAsync(_awaitableEventArgs))
+            using (ExecutionContext.SuppressFlow())
```

Comment:
Can the follow already be suppressed when it gets here? (in which case this call would throw)

Reply:
Unlikely but I can add the crazy code. Regardless, I’m sad this is per call instead of per construction

Reply:
The flow can't be suppressed here, this API is called from a single place in the code base.